machine.go

```diff
@@ -880,3 +869,30 @@ func (m *Machine) waitForSocket(timeout time.Duration, exitchan chan error) erro
 		}
 	}
 }
+
+// Set up a signal handler and pass INT, QUIT, and TERM through to firecracker
+func (m *Machine) setupSignals() {
+	if m.Cfg.DisableSignals {
```

Comment:
How about instead of a `DisableSignals` flag, we have a strategy that allows for setting signals up. This strategy could be something like

```go
type BootstrapSignals func() error

type Machine struct {
    // machine field here
    BootstrapSignals BootstrapSignals
}
```

Then the default would be the `setupSignals` method you already have, but you can overwrite it with whatever you want.

Reply:
It seems host apps need a way to control the list of signals to forward rather than inject new behavior. A bootstrap function is heavy handed for apps that just need to change which signals are forwarded since Machine doesn't export the command, so everything needs to be re-implemented. And for apps that don't want forwarding, an empty list seems clearer than a noop bootstrap function. 

I think exposing an optional list of signals to handle (which defaults to the current list) would make a cleaner API than injecting a function. What do you think?